### ISSUE TITLE
Update to Ghidra 11.1

### ISSUE THREAD

**Issue by toolCHAINZ** [toolCHAINZ/jingle#1]
This is currently blocked.

In Ghidra 11.1, `sleigh` was updated to support compressed architecture modules. This feature brings along a mandatory dependency on `zlib`.

So, to update, need to either:

* Add a build.rs step that ensures zlib is installed and adds it to the linking path
* Vendor it in with the other C++ and use the "local zlib" build option.
* Maybe there's a way to do option 2 but link it to [libz-sys](https://crates.io/crates/libz-sys) instead

**Comment by VPaulV** [comment]
Good day!

I would greatly appreciate it if you could update your bindings to Ghidra 11.1.1 when have time. I was about to write another wrapper around Sleigh because none of the existing solutions seemed satisfactory. However, I was lucky to discover your bindings, which seems to be perfect. Thank you for sharing!

**Comment by toolCHAINZ** [comment]
Hey! Happy it looks useful to you! I'm half way through puzzling this out. I got it building on the dev branch using `libz-sys` but am having some runtime issues. It looks like sleighs document parser expects the file to be decompressed already and I have yet to find a good example in the sleigh codebase of how/when it expects decompression to be done. Hopefully will get that figured out today.

**Comment by toolCHAINZ** [comment]
Gah GitHub is being flaky on my phone. Glad this looks useful to you! Working on this (and have it building), but the sla parsing expects a decompressed spec, so I need to work out how/when they intend decompression to be done.

**Comment by toolCHAINZ** [comment]
Ok, I think I've gotten it working. It's up on `dev`. Need to test it with my other code still, but my unit tests pass, so it's at least somewhat working.

I'm not going to bother with any historical support, so I'm just going to assume ghidra version >= 11.1.0 from here on out. I have not tried these changes on older versions. I copied how the sleigh example loads architectures now so hopefully the historical support is at least no worse than whatever ghidra itself is doing.

Closing this as of [adaa1b8](https://github.com/toolCHAINZ/jingle/commit/adaa1b869507ea66d2fdfcfe573581f046fb60b5)
